File: Linear Search/linear_search.py
# -*- coding: utf-8 -*-
"""
Created on Fri Sep 28 21:17:54 2018

@author: Nazrul
"""

def linear_search(aList, sizeOfList, aVar):
    result = False
    for i in range(sizeOfList):
        if aList[i] == aVar:
            print(aVar, "at " + str(i) + " Matched")
            result = True
        else:
            print(aVar, "at " + str(i) + " Not matched")
    return result


integers = [10,20,30,40,50,60,70,80,90,100]
find = 20

if linear_search(integers, len(integers), find):
    print("Found")
else:
    print("Not Found")

    

File: Linear Search/linear_search.cpp
#include <iostream>
using namespace std;

bool linear_search(int List[], int SizeOfList, int var){
    bool result = false;
    for(int i = 0; i < SizeOfList; i++){
        if (List[i] == var) {
            cout << var << " at " << i << " Matched" << endl;
            result = true;
        }
        else{
            cout << var << " at " << i << " Not matched" << endl;
        }
    }

    return result;
}

int main()
{
    int IntList[] = {10, 20, 30, 40, 50, 60, 70, 80, 90, 100};
    int Find = 20;

    if(linear_search(IntList, sizeof(IntList)/4, Find)){
        cout << "Found" << endl;
    }
    else{
        cout << "Not Found" << endl;
    }

    return 0;
}
